### DISCUSSION TITLE
Warning: React does not recognize the isValid prop on a DOM element.

### DISCUSSION THREAD

**Discussion by NicoleHoerner** [neuefische/questions#731] [Web] [answered]
Hello,
I implemented a form and some styling/functions to t I get an error message that I am not able to solve and I am not sure if it matters if it persists.

ERROR:

(client.js:1 Warning: React does not recognize the `isValid` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isvalid` instead. If you accidentally passed it from a parent component, remove it from the DOM element.)

Here is my [PR](https://github.com/NicoleHoerner/Capstone-Project-Flight-Tracking/pull/10)

Has it something to do with this part?:
export const InputLabel = styled.span`
  position: absolute;
  left: 0;
  top: 0px;
  transform: translateY(-4px);
  margin-left: 10px;
  padding: 10px;
  pointer-events: none;
  font-size: 0.6rem;
  color: #000;
  text-transform: uppercase;
  transition: 0.5s;
  letter-spacing: 3px;
  border-radius: 8px;
  transform: ${(props) =>
    props.isFocused || props.isValid
      ? "translateX(113px) translateY(-15px)"
      : "translateY(-4px)"};
  font-size: ${(props) =>
    props.isFocused || props.isValid ? "0.8em" : "12px"};
  padding: ${(props) =>
    props.isFocused || props.isValid ? "5px 10px" : "10px"};
  background: ${(props) =>
    props.isFocused || props.isValid ? "#000" : "transparent"};
  letter-spacing: ${(props) =>
    props.isFocused || props.isValid ? "0.2em" : "3px"};
  color: ${(props) => (props.isFocused || props.isValid ? "#fff" : "#cbd5e1")};
  border: ${(props) => (props.isFocused || props.isValid ? "2px" : "initial")};
`;

export const InputField = styled.input`
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 200px;
  padding: 8px;
  border: 2px solid
    ${(props) => (props.isFocused || props.isValid ? "#cbd5e1" : "initial")};
  border-radius: 8px;
  &:focus {
    border-color: #cbd5e1;
  }
  &:valid {
    border-color: #cbd5e1;
  }
`;

What have I already tried:

I have already searched for this error online and found solutions that didn't help me.
E.g. change the styling 
import styled, { css } from "styled-components";

export const InputLabel = styled.span`
  // Add common styles here
  position: absolute;
  left: 0;
  top: 0px;
  margin-left: 10px;
  padding: 10px;
  pointer-events: none;
  font-size: 0.6rem;
  color: #000;
  text-transform: uppercase;
  transition: 0.5s;
  letter-spacing: 3px;
  border-radius: 8px;

  ${(props) =>
    props.isFocused || props.isValid
      ? css`
          transform: translateX(113px) translateY(-15px);
          font-size: 0.8em;
          padding: 5px 10px;
          background: #000;
          letter-spacing: 0.2em;
          color: #fff;
          border: 2px;
        `
      : css`
          transform: translateY(-4px);
          font-size: 12px;
          padding: 10px;
          background: transparent;
          border: initial;
        `}
`;

it didn't work.

Here is a post that might help but I don't know how.
https://legacy.reactjs.org/warnings/unknown-prop.html

Thanks for any help!
Regards, Nicole

**Comment by Satttoshi** [comment, marked as answer]
You can quickfix this with renaming the prop to `$isValid`
